packages/loader/container-loader/src/container.ts

```diff
@@ -1350,28 +1378,30 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
         const protocolHandler = await this.initializeProtocolState(
             attributes,
-            members,
-            proposals,
-            values);
+            quorumSnapshot,
+            protocolHandlerBuilder);
 
         return protocolHandler;
     }
 
     private async initializeProtocolState(
         attributes: IDocumentAttributes,
-        members: [string, ISequencedClient][],
-        proposals: [number, ISequencedProposal, string[]][],
-        values: [string, any][],
+        quorumSnapshot: IQuorumSnapshot,
+        protocolHandlerBuilder: ProtocolHandlerBuilder | undefined,
     ): Promise<IProtocolHandler> {
-        const protocol = new ProtocolOpHandlerWithClientValidation(
-            attributes.minimumSequenceNumber,
-            attributes.sequenceNumber,
-            attributes.term,
-            members,
-            proposals,
-            values,
-            (key, value) => this.submitMessage(MessageType.Propose, { key, value }),
-        );
+        const protocol = protocolHandlerBuilder !== undefined ?
```

Comment:
Feels like code would be more readable if function argument would have default value, i.e.
```ts
protocolHandlerBuilder: ProtocolHandlerBuilder = (...args) => new ProtocolHandler(...args),
```